framework_tvb/tvb/interfaces/web/controllers/flow_controller.py

```diff
@@ -305,23 +305,25 @@ def execute_post(self, project_id, submit_url, step_key, algorithm, **data):
                 raise InvalidFormValues("Invalid form inputs! Could not fill algorithm from the given inputs!",
                                         error_dict=form.get_errors_dict())
 
-            adapter_instance.submit_form(form)
-
-            if issubclass(type(adapter_instance), ABCDisplayer):
-                adapter_instance.current_project_id = project_id
-                adapter_instance.user_id = common.get_logged_user().id
-                result = adapter_instance.launch(view_model)
-                if isinstance(result, dict):
-                    return result
-                else:
-                    common.set_error_message("Invalid result returned from Displayer! Dictionary is expected!")
-                return {}
-
-            result = self.operation_services.fire_operation(adapter_instance, common.get_logged_user(),
+            dts_imported = self.operation_services.import_dts(view_model, project_id)
```

Comment:
This is a very odd change. Why? The export was functioning before, can't we reuse the mechanism?

Reply:
In order to test what happens on the mechanism before, the line in question can be commented and dts_imported given the value False. The change was necessary because if we try with the old version a additional empty TVB importer operation will be created dew to the flow(this could not be ommited whit the old flow, tested and researched during debug), also the old flow seems to ommit the time information on the Data Structure page in UI.

Reply:
A TVBImport operation will be needed, right? This is what we execute.

ViewModel files should not be exported in this flow, so all linked DTs will not have any operation information with them, thus at import time, we execute a TVBImport.

Reply:
The export is not the problem here, the import is...when we uploas a file in TVB HDF5/ZIP importer overlay the first thing that is created is a TVBImporterModel and it creates a operation for that model, and then the flow continues as in the case of the import project creating a new operation for each folder in the zip file we want to import -> leaving us with the first operation folder containing only the TvbImporterModel and the rest are imported normaly...

Reply:
I think the "imported normally like at import project" is the problem here.
Why not export the dts as a zip with no extra folders. Then import with a tVBimporter operation created.

Reply:
I tested with the mentioned solution for the export (all h5 in the zip file no subfolder), and with the current implementation of the TVB HDF5/ZIP importer, we still have the problem that a extra operation with only TVBImporterModel will be first created, and then all the other h5's will be imported in the same operation... tested with zip containing conn, surface and region mapping, the result looked a bit unusual/wierd to me in the operations dashboard... is this what we want? or separate operations for each file in the zip?

Reply:
I think just one operation will be ok.
So it is left to put all imported dts to the first operation that you mention (the one that you Say it is now empty)